pageserver/src/tenant.rs

```diff
@@ -1068,6 +1066,19 @@ impl Tenant {
         let remote_startup_data = match &remote_client {
             Some(remote_client) => match remote_client.download_index_file().await {
                 Ok(index_part) => {
+                    let index_part = match index_part {
+                        MaybeDeletedIndexPart::IndexPart(index_part) => index_part,
+                        MaybeDeletedIndexPart::Deleted => {
+                            info!("is_deleted is set on remote, proceeding to remove local data");
+                            std::fs::remove_dir_all(
+                                self.conf.timeline_path(&timeline_id, &self.tenant_id),
+                            )
+                            .context("remove_dir_all")?;
+
+                            return Ok(());
+                        }
+                    };
+
                     let remote_metadata = index_part.parse_metadata().context("parse_metadata")?;
                     Some(RemoteStartupData {
                         index_part,
```

Comment:
What if we delete a timeline with remote storage, then restart without remote storage?

I realize this is an unrealistic use case, but, I think it highlights the more fundamental issue that we don't persist the deletedness of a timeline locally.

Reply:
Added a TODO comment.